src/topics/resolver/ens-resolver.ts

```diff
@@ -20,30 +22,84 @@ export class EnsResolver extends GraphQLProvider implements IResolver {
       : (this.provider = ethers.getDefaultProvider());
   }
 
-  public async resolve(ensData: string): Promise<string> {
+  public async resolve(accounts: FetchedData): Promise<FetchedData> {
+    let resolvedAccounts: FetchedData = {};
+
+    const accountsArray = Object.entries(accounts);
+
+    const resolvedAccountsArray = await withConcurrency(
+      accountsArray,
+      this.resolveENSAccounts,
+      {
+        batchSize: 50,
+        concurrency: 10,
+      }
+    );
+
+    resolvedAccounts = resolvedAccountsArray.reduce(
+      (accumulator, currentObject) => {
+        return { ...accumulator, ...currentObject };
+      },
+      {}
+    );
+
+    return resolvedAccounts;
+  }
+
+  private resolveENSAccounts = async (
+    accounts: [string, BigNumberish][]
+  ): Promise<FetchedData> => {
+    const ensNames = accounts.map((item) => item[0]);
+    const domains = await this.resolveEnsHandlesQuery(ensNames);
+
+    // if it didn't resolve all the accounts, throw an error
+    if (domains.length < accounts.length) {
```

Comment:
You should compare domains length to ensNames length, and return ensName instead of domain in case of error since domains are successful outputs while ensNames are the inputs